staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -368,29 +366,29 @@ func newTestableController(config TestableConfig) *configController {
 	return cfgCtlr
 }
 
-func (cfgCtlr *configController) Run(stopCh <-chan struct{}) error {
-	defer utilruntime.HandleCrash()
-
-	// Let the config worker stop when we are done
-	defer cfgCtlr.configQueue.ShutDown()
+func (cfgCtlr *configController) Start(ctx context.Context) error {
+	go func() {
+		<-ctx.Done()
+		// Let the config worker stop when we are done
+		klog.Info("Shutting down API Priority and Fairness config worker")
+		cfgCtlr.configQueue.ShutDown()
+	}()
 
 	klog.Info("Starting API Priority and Fairness config controller")
-	if ok := cache.WaitForCacheSync(stopCh, cfgCtlr.plInformerSynced, cfgCtlr.fsInformerSynced); !ok {
-		return fmt.Errorf("Never achieved initial sync")
+	if ok := cache.WaitForCacheSync(ctx.Done(), cfgCtlr.plInformerSynced, cfgCtlr.fsInformerSynced); !ok {
```

Comment:
the docs say `Start()` doesn't block, but this _does_ make it block until informers sync or context is canceled, right?

Reply:
What the comment about not blocking is trying to say is that this is not "Run" --- this method does not block until the controller is done running. This is in contrast to some other methods in the Kubernetes code base that are named "Start" but are really a _Run_ method.

Reply:
Fixed the comment in the interface.